src/corecel/data/Collection.hh

```diff
@@ -127,6 +127,53 @@ using ItemId = OpaqueId<T, size_type>;
 template<class T, class Size = size_type>
 using ItemRange = Range<OpaqueId<T, Size>>;
 
+//---------------------------------------------------------------------------//
+/*! Access data in a Range<T2> with an index of type T1
+ *
+ * Here, T1 and T2 are expected to be OpaqueId types
+ */
+template<class T1, class T2>
+class ItemMap
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using key_type = T1;
+    using mapped_type = T2;
+    //!@}
+
+  public:
+    //// CONSTRUCTION ////
+
+    ItemMap() = default;
+
+    //! Contruct from an exising Range<T2>
+    ItemMap(Range<T2> range) : range_(range)
+    {
+        static_assert(detail::IsOpaqueId<T1>::value, "T1 isn't opaque ID");
+        static_assert(detail::IsOpaqueId<T2>::value, "T2 isn't opaque ID");
```

Comment:
These should probably be at the class level, not the constructor level. I'd put them right after the opening brace on the class, since they're constraints on the class template parameter rather than the constructor.